### ISSUE TITLE
Reading Week project

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/master-reference#901]
I would like us to consider setting a Reading Week project to consolidate the first 8 weeks' learning with a small individual project (we could perhaps offer 3-4 project ideas) that we will recommend for students to complete solo, in order for them to have the experience of building all parts of a project without assistance. The project would not be presented in Week 10, but it could be recommended as a first portfolio project for each student's CV.

**Why?** 
- Working in groups, not all students will have had the opportunity to work on all parts of the stack
- While the course is primarily group-based, there is also value in working solo on a project--which is not something that students will get a chance to do otherwise during the programme
- It will give anyone who feels that have fallen behind a bit in the first 8 weeks an opportunity to do some revision before the 2nd half of the programme

**Comment by astroash** [comment]
IMO additional projects are not the best use of most students time during reading week. There is a lot of time spent on rapid building in the first half, which is a great way to get an overview of how full stack apps apps fit together, but leaves big gaps in students technical knowledge. I think it would be more appropriate for us to create a list of recommended reading/workshops for students to look at in their own time, and encourage students to revisit workshops they struggled with/did not finish. Off the top of my head areas that I think would be useful are:
- array methods
- closure
- es6 features
- promises/ async await
- web app architecture

If students are able to brush up on some of their js understanding during the break I think they will get more out of the second half of the course.

**Comment by FarahZaqout** [comment]
I think it might be worthwhile to consider having both options available because I'd imagine in each cohort there would be people who are keeping up well during the 8 weeks and would benefit more if they focus on improving their javascript skills, but there would also be people who might feel that they still need to put some effort to better understand the stack, sometimes rightly so. For the latter, having a reading list available that they can get back to will give them the best of both worlds.

**Comment by lucymk** [comment]
Being as there are going to be people of both types on every cohort, maybe the best would be to find half an hour in Week 8 to have some sort of workshop/chat to help students identify what the best use of their time would be. Then from this point have recommended reading/exercises etc for consolidation, or suggested project outline.

**Comment by yasminhillis** [comment]
A solo project during the reading week is definitely a great idea. We, in Gaza, have experienced with it with both the Fourth and Fifth cohorts. We, Ghada @ibraghada, Ahamed @aajour  and myself, announced for a mini-hackathon and asked the students to deliver a Reddit, using the technologies learned during the eight weeks. Then, we asked mentors to review the code. And now we are in the process of negotiating the winner. This is supposed to help students as it allows them to realize their weak areas- the parts that might be covered by teammates in a group project. But it depends on the student's level of awareness and their readiness to face what they feel unsure about. Tbh, I didn't feel that students faced their fears when it comes to coding. **Maybe because we were doing the mini-hackathon during the Eid holiday**. 
Some students wanted to do some further reading/ going back to workshops and we were totally fine with it. I mean, we gave the students the space to do what felt right for them
Despite our experience with reading week project, I really encourage you to give it a shot, **if students' show an interest in it**.
I suggest, if you decided to go for solo projects already, is to: **first** ask all students what they are more inclined to do during this week. **Second,**: suggest further-reading lists/ ideas for projects accordingly. **Third**: assign mentors to review code. **Fourth:** Announce the winner. **Fifth:** Hold One-on-Ones with students to get feedback on how to better the process in the future.

**Comment by arrested-developer** [comment]
It's worth remembering that a lot of consolidation happens just by stepping away from your learning. It would always amaze me how many of my school students would struggle with a new concept in a lesson, then come back in the next day having slept on it and be much more confident.

I'd also like to point out that for many (most? all?) of us, the past 8 weeks were so intense that we desperately needed to use reading week to slow down and relax. Personally, I really appreciated the force with which @lucymk insisted we shouldn't feel the need to code during week 8. I think introducing a project during this week risks placing a new pressure on students that might not be positive.

@yasminhillis The mini hackathon sounds awesome! I wonder though if making this a competition has prevented students from confronting areas they felt weak in?

I'm in favour of a recommended reading list, especially if FAC bought in some real paper books (budget permitting) and we could do this away from screens 💻

**Comment by ashatat** [comment]
I agree that a solo project is a good Idea to help students get benefits of working on all project aspects by themselves, and figure out what they need to work on it,
also, I agree that the first 8 weeks were so intense, so some may need a refreshing time

I would suggest that to let the student decide what is their best for them and give them a small project that they could work on it but not complicated to make them rush into finishing it without learning some stuff

**Comment by shiryz** [comment]
I have to agree with @arrested-developer, reading week is the best time to take a step back from coding and refresh your mental energies.
As it has been mentioned before the first 8 weeks can be pretty draining, having a week off is where students get the option to relax for a bit, they might feel inclined to do a project if offered that option.
This time around in Nazareth we had no reading week, which was replaced by react week, both the mentors and students could have used this to recharge :relaxed: :sleeping:

**Comment by m4v15** [comment]
#925
